roles/autossl/defaults/main.yml

```diff
@@ -1,2 +1,50 @@
 ---
-# defaults file for autossl
+
+## Deployment
+replicas: 1
+image_name: "quay.io/3scale/autossl"
+image_version: "v1.0.0"
+image_pull_secret_name: "quay-pull-secret"
+acme_staging: ""
+contact_email: "3scale-operations@redhat.com"
+proxy_endpoint: "https://multitenant-admin.dev.3sca.net"
+storage_adapter: "redis"
+redis_host: "dev-saas-autossl-ec-redis.ng.0001.use1.cache.amazonaws.com"
+redis_port: "6379"
+verification_endpoint: "https://multitenant-admin.dev.3sca.net/swagger/spec.json"
```

Comment:
Maybe this very environment specific values (redis_host, contact_email, verification_endpoint, proxy_endpoint) shouldn't have defaults?

Reply:
It is ansible, it is a good practice to have defaults valus to all vars, so once it is executed, there is no failed task telling that var "x" is undefined (ansible operator executes a sequential playbook, if one middle task fails, next tasks are not executed unless you put the ignore_errors on the task). 

In addition, I think default values are kind of documentation of the expected value.

Actually almost all vars are fake (the ones that yes or yes need to be passed on CR, different endpoints...), except the email, which shouldn'tt change among our different installations, avoiding to specify then on the CR.

What I can do is to not set our real mail on default value, and put a fake one like the different endpoints.

Also, by the moment I haven't configured CRD validation, but on CRD validation you can configure required values (so if you don't specify it on CR, k8s API will tell you you miss a CR field, and don't apply it), so we can decide which are the required vars and specify them on CRD.

What do you think about my opinion on default values, email default value, and CRD validation with required vars? @roivaz

Reply:
It would be nice to have them as examples in a documentation file, but having it as default can cause issues. As you can forgot to set a configuration that is required and won't fail, as will default to the dev configuration.

Reply:
Fixed https://github.com/3scale/saas-operator/pull/2/commits/94edc1d3ea1f64915a4339d02c7563d8a5c5e235